base/array.jl

```diff
@@ -2467,3 +2467,192 @@ end
 
 intersect(itr, itrs...) = _shrink(intersect!, itr, itrs)
 setdiff(  itr, itrs...) = _shrink(setdiff!, itr, itrs)
+
+map_unrolled(call, variables::Tuple{}) = ()
+map_unrolled(call, variables) =
+    call(first(variables)), map_unrolled(call, tail(variables))...
```

Comment:
I think `Base.map` does this already.

Reply:
Yup, but I wanted to make super duper sure that it was going to unroll, not iterate

Reply:
I put the unrolled operations into a separate mini-module